src/NosCore.GameObject/Providers/UpgradeService/UpgradeService.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using ChickenAPI.Packets.ClientPackets.Inventory;
+using ChickenAPI.Packets.Enumerations;
+using ChickenAPI.Packets.Interfaces;
+using ChickenAPI.Packets.ServerPackets.Inventory;
+using ChickenAPI.Packets.ServerPackets.Shop;
+using ChickenAPI.Packets.ServerPackets.UI;
+using NosCore.Core;
+using NosCore.Core.I18N;
+using NosCore.Data.Enumerations.I18N;
+using NosCore.GameObject.ComponentEntities.Extensions;
+using NosCore.GameObject.Helper;
+using NosCore.GameObject.Networking.ClientSession;
+using NosCore.GameObject.Providers.InventoryService;
+using NosCore.GameObject.Providers.ItemProvider.Item;
+using Serilog;
+
+namespace NosCore.GameObject.Providers.UpgradeService
+{
+    public class UpgradeService: IUpgradeService
+    {
+        private readonly ILogger _logger;
+        private readonly Dictionary<UpgradePacketType,
+            Func<ClientSession, InventoryItemInstance, InventoryItemInstance, InventoryItemInstance>> PacketsFunctions;
+
+        public UpgradeService(ILogger logger)
+        {
+            _logger = logger;
+            PacketsFunctions = new Dictionary<UpgradePacketType, Func<ClientSession, InventoryItemInstance, InventoryItemInstance, InventoryItemInstance>>
+                {
+                    { UpgradePacketType.SumResistance, new Func<ClientSession, InventoryItemInstance, InventoryItemInstance, InventoryItemInstance>(Sum) }
+                };
+        }
+
+        public void HandlePacket(UpgradePacketType type, ClientSession clientSession, InventoryItemInstance item1, InventoryItemInstance item2)
+        {
+            PacketsFunctions[type].DynamicInvoke(clientSession, item1, item2);
+        }
+
+        #region Sum
+
+        private InventoryItemInstance Sum(ClientSession clientSession, InventoryItemInstance item, InventoryItemInstance itemToSum)
```

Comment:
ClientSession shouldn’t be here

Reply:
What can I do instead ? Can you explain what you want ?